packages/plugin-ma-tracking/src/advanced-campaign-tracker.ts

```diff
@@ -0,0 +1,43 @@
+import { Campaign, CampaignTrackerOptions } from '@amplitude/analytics-types';
+import { CampaignTracker } from '@amplitude/analytics-client-common';
+
+export class AdvancedCampaignTracker extends CampaignTracker {
+  resetSessionOnNewCampaign: boolean;
+
+  private _currentCampaign!: Campaign;
+  private _isNewCampaign = true;
+  ready: Promise<unknown>;
+
+  constructor(apiKey: string, options: CampaignTrackerOptions) {
+    super(apiKey, options);
+
+    this.resetSessionOnNewCampaign = options.resetSessionOnNewCampaign ?? Boolean(options.trackNewCampaigns);
+    this.ready = new Promise((resolve) => resolve(this.refreshCampaignState()));
```

Comment:
Where will this be used?

Reply:
When `getCurrentState` is called. It is to make sure `isNewCampaign` and `currentCampaign` are ready.